repositories/city.repo.go

```diff
@@ -0,0 +1,105 @@
+package repositories
+
+import (
+	"errors"
+	"fmt"
+	"github.com/gofiber/fiber/v2"
+	"gorm.io/gorm"
+	"net/http"
+	"project_weather/models"
+)
+
+type CityRepository struct {
```

Comment:
This class handles both the DB queries and API requests. There is a concept in software engineering for making layered applications e.g. From outside to inside - Network handlers - Business logic - DB and persistence (or as many as there are needed) [https://tutorialedge.net/software-eng/onion-architecture/](https://tutorialedge.net/software-eng/onion-architecture/)